File: imstargg-frontend/src/model/BattlePlayer.ts
import {BattlePlayerBrawler} from "@/model/BattlePlayerBrawler";
import {SoloRankTier} from "@/model/enums/SoloRankTier";

export interface BattlePlayer {
  tag: string;
  name: string;
  soloRankTier?: SoloRankTier;
  brawler: BattlePlayerBrawler;
}

File: core/core-domain/src/main/java/com/imstargg/core/domain/player/BattlePlayer.java
package com.imstargg.core.domain.player;

import com.imstargg.core.domain.BrawlStarsTag;
import com.imstargg.core.enums.SoloRankTier;
import jakarta.annotation.Nullable;

public record BattlePlayer(
        BrawlStarsTag tag,
        String name,
        @Nullable SoloRankTier soloRankTier,
        BattlePlayerBrawler brawler
) {
}
